src/components/routes/chat/ChatPanel.tsx

```diff
@@ -124,13 +131,37 @@ enum ViewType {
 	Video = 'video'
 }
 
+const useTyping = (channelID: string) => {
+	const [isTyping, setIsTyping] = useState(false);
+	const [timer, setTimer] = useState<NodeJS.Timeout>(() => setTimeout(() => console.log('Typing initialised'), 0));
+	const currentUserID = useSelector(selectMe)?.id;
+
+	useEffect(() => {
+		const listener = client.gateway?.on(GatewayPacketType.GatewayTyping, data => {
+			console.log(timer);
+			if (data.channelID === channelID && data.userID !== currentUserID) {
+				console.log('in if');
+				setIsTyping(true);
+				clearTimeout(timer);
+				setTimer(setTimeout(() => setIsTyping(false), 5000));
+			}
+		});
+		return () => {
+			listener?.destroy();
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [timer]);
```

Comment:
The timer never changes for some reason and I can't figure out why. When you put a `console.log(timer);` inside the listener, it always prints the same value so the timer never actually gets rewritten. @amishshah do you have any idea why this is the case?